sys/net/dsm/dsm.c

```diff
@@ -61,9 +64,7 @@ dsm_state_t dsm_store(sock_dtls_t *sock, sock_dtls_session_t *session,
     }
 
     prev_state = session_slot->state;
-    if (session_slot->state != SESSION_STATE_ESTABLISHED) {
-        session_slot->state = new_state;
-    }
+    _set_state(new_state, session_slot);
```

Comment:
Once implemented, I'd suggest you add a check on the returned value to verify that the state change was valid.

Reply:
For `dsm_store` it doesn't matter if the state was changed. Calling it without a valid state change is ok (e.g. even in the normal CoAP/DTLS flow it gets called with`SESSION_STATE_HANDSHAKE` while `SESSION_STATE_ESTABLISHED` is already set). So, I'm not sure what I would include in that check. It would just continue the same way with or without a changed state. That's why I didn't include any check in the first place.